homeassistant/components/thread/dataset_store.py

```diff
@@ -0,0 +1,150 @@
+"""Persistently store thread datasets."""
+from __future__ import annotations
+
+import dataclasses
+from datetime import datetime
+from typing import Any, cast
+
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.storage import Store
+from homeassistant.util import dt as dt_util, uuid as uuid_util
+
+from . import tlv_parser
+from .const import DOMAIN
+
+DATA_STORE = "datasets"
+STORAGE_KEY = "thread.datasets"
+STORAGE_VERSION_MAJOR = 1
+STORAGE_VERSION_MINOR = 1
+SAVE_DELAY = 10
+
+
+@dataclasses.dataclass(frozen=True)
+class DatasetEntry:
+    """Dataset store entry."""
+
+    preferred: bool
+    source: str
+    tlv: str
+
+    created: datetime = dataclasses.field(default_factory=dt_util.utcnow)
+    id: str = dataclasses.field(default_factory=uuid_util.random_uuid_hex)
```

Comment:
Let's use a ulid instead of uuid. Or should we derive an ID from the source (with _2 etc added if it already exists)